gateway/ln-gateway/src/lib.rs

```diff
@@ -403,6 +403,7 @@ impl Gateway {
         Gateway::new(
             Arc::new(GatewayLightningBuilder {
                 lightning_mode: opts.mode,
+                gateway_db: gateway_db.clone(),
```

Comment:
@tvolk131 this unfortunately might break some of your proposed refactors/abstractions

Reply:
So be it